src/decorators/index.ts

```diff
@@ -45,25 +52,42 @@ const decodeAndVerify = (tokenJwtString: string) => {
   }
 };
 
+const verifyEgoApiKey = async (uuidString: string) => {
+  const EGO_URL = config.getConfig().egoUrl();
+  const EGO_CLIENT_ID = config.getConfig().egoClientId();
+  const EGO_CLIENT_SECRET = config.getConfig().egoClientSecret();
+  const auth = 'Basic ' + Buffer.from(EGO_CLIENT_ID + ':' + EGO_CLIENT_SECRET).toString('base64');
+
+  const response = await fetch(`${EGO_URL}/o/check_api_key?apiKey=${uuidString}`, {
+    method: 'post',
+    headers: { 'Content-Type': 'application/json', authorization: auth },
+  });
+  const token = await response.json();
+  return token;
+};
+
 const hasScope = (scopes: string[], token: any) => {
   if (
-    !token.context ||
-    !token.context.scope ||
-    token.context.scope.filter((s: string) => scopes.indexOf(s) >= 0).length === 0
+    (token.context &&
+      (!token.context.scope ||
+        token.context.scope.filter((s: string) => scopes.indexOf(s) >= 0).length === 0)) ||
+    (token.scope && token.scope.filter((s: string) => scopes.indexOf(s) >= 0).length === 0)
   ) {
     return false;
   }
   return true;
 };
```

Comment:
I'm suggesting a re-write of this method so that it no longer accepts `any` as an input. Making this accept just the provided scopes from the token makes its logic very straightforward, and pushes all teh token parsing into the code that is already handling tokens.

```TypeScript

/**
 * Confirms if the at least one of the required scopes has been provided
 * @param requiredScopes at least one of these scopes must be provided
 * @param providedScopes scopes provided in auth header
 * @returns {boolean} true if provided scopes contains at least one of the requiredScopes
 */
const hasScope = (requiredScopes: string[], providedScopes: string[]): boolean => 
  requiredScopes.some(scope => providedScopes.includes(scope));
  
  ```

Reply:
Updated